File: app/src/androidTest/java/com/sakuna63/tumbin/application/contract/presenter/login/OauthHelperTestAndroid.java
package com.sakuna63.tumbin.application.contract.presenter.login;

import android.support.test.filters.SmallTest;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

@RunWith(AndroidJUnit4.class)
@SmallTest
public class OauthHelperTestAndroid {

    @Test
    public void extractVerifier_includeOauthVerifier_verifier() throws Throwable {
        String expected = "verifier";
        String inputUrl = "http://example.com?oauth_verifier=" + expected;

        String actual;
        {
            actual = OauthHelper.extractVerifier(inputUrl);
        }

        assertThat(actual, is(expected));
    }

    @Test
    public void extractVerifier_notIncludeOauthVerifier_null() throws Throwable {
        String expected = null;
        String inputUrl = "http://example.com";

        String actual;
        {
            actual = OauthHelper.extractVerifier(inputUrl);
        }

        assertThat(actual, is(expected));
    }

    @Test
    public void extractVerifier_notUrlForm_null() throws Throwable {
        String expected = null;
        String inputUrl = "input";

        String actual;
        {
            actual = OauthHelper.extractVerifier(inputUrl);
        }

        assertThat(actual, is(expected));
    }
}

File: app/src/androidTest/java/com/sakuna63/tumbin/application/contract/presenter/login/OauthHelperTestAndroid.kt
package com.sakuna63.tumbin.application.contract.presenter.login

import android.support.test.filters.SmallTest
import android.support.test.runner.AndroidJUnit4
import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.MatcherAssert.assertThat
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@SmallTest
class OauthHelperTestAndroid {

    @Test
    @Throws(Throwable::class)
    fun extractVerifier_includeOauthVerifier_verifier() {
        val expected = "verifier"
        val inputUrl = "http://example.com?oauth_verifier=" + expected
        val actual = OauthHelper.extractVerifier(inputUrl)
        assertThat(actual, `is`(expected))
    }

    @Test
    @Throws(Throwable::class)
    fun extractVerifier_notIncludeOauthVerifier_null() {
        val expected: String? = null
        val inputUrl = "http://example.com"
        val actual = OauthHelper.extractVerifier(inputUrl)
        assertThat(actual, `is`<String>(expected))
    }

    @Test
    @Throws(Throwable::class)
    fun extractVerifier_notUrlForm_null() {
        val expected: String? = null
        val inputUrl = "input"
        val actual = OauthHelper.extractVerifier(inputUrl)
        assertThat(actual, `is`<String>(expected))
    }
}